tests/integration/test_storage_kafka/test.py

```diff
@@ -1535,7 +1537,9 @@ def test_kafka_flush_by_block_size(kafka_cluster):
         messages.append(json.dumps({'key': 0, 'value': 0}))
     kafka_produce('flush_by_block_size', messages)
 
-    time.sleep(1)
+    # Wait for Kafka engine to consume this data
+    while 1 != int(instance.query("SELECT count() FROM system.parts WHERE database = 'test' AND table = 'view' AND name = 'all_1_1_0'")):
```

Comment:
worth limit number of attempts?

Reply:
Maybe but they are already limited by `@pytest.mark.timeout(600)`